src/plugins/internalnotification/internalnotification.c

```diff
@@ -0,0 +1,263 @@
+/**
+ * @file
+ *
+ * @brief Source for internalnotification plugin
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#include "internalnotification.h"
+
+#if PLUGIN_INTERNALNOTIFICATION_VERBOSE
+#include <stdio.h>
+#endif
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include <kdbhelper.h>
+
+struct _KeyRegistration
+{
+	char * name;
+	int * variable;
+	struct _KeyRegistration * next;
+};
+typedef struct _KeyRegistration KeyRegistration;
+
+struct _ListPointer
+{
+	KeyRegistration * head;
+	KeyRegistration * tail;
+};
+typedef struct _ListPointer ListPointer;
+
+
+static KeyRegistration * elektraInternalnotificationAddNewRegistration (ListPointer * listPointer)
+{
+	KeyRegistration * item = elektraMalloc (sizeof *item);
```

Comment:
it's legal, but better stop talking about it, otherwise you could start a war ;)